packages/ui/src/components/va-modal/VaModal.vue

```diff
@@ -1,30 +1,20 @@
 <template>
-  <teleport v-if='valueComputed' :to="attachElement" :disabled="disableAttachment">
-    <div class="va-modal">
-      <modal-element
-        name="va-modal__overlay--transition"
-        :isTransition="!$props.withoutTransitions"
-        @click="onOutsideClick"
-        appear
-      >
+  <teleport :to="attachElement" :disabled="disableAttachment">
+    <modal-element name="va-modal" :isTransition="!$props.withoutTransitions" appear>
```

Comment:
As I can see here used nested transition. Vue listens for transition events (transitioned or animationend). Since there is no transition property on `.va-modal` vue can not determinate transition duration. 

So, it's required `Transition` child to have transition property set, or you can manually provide duration to transition component.

```suggestion
    <modal-element name="va-modal" :isTransition="!$props.withoutTransitions" appear :duration="300">
```

Then vue will set animation classes for specific duration and nested animations (`.va-modal__container`) will not ended in 0.00001 sec :)

Reply:
Previously, animation duration was taken from `va-modal-transition` scss mixin. I think it's better to have animation duration prop 😁.